File: php/features.php
<?php
declare(strict_types=1);

// JmespathFree SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class JmespathFreeFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new JmespathFreeBaseFeature();
            case "test":
                return new JmespathFreeTestFeature();
            default:
                return new JmespathFreeBaseFeature();
        }
    }
}

File: rb/features.rb
# JmespathFree SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module JmespathFreeFeatures
  def self.make_feature(name)
    case name
    when "base"
      JmespathFreeBaseFeature.new
    when "ratelimit"
      JmespathFreeRatelimitFeature.new
    when "retry"
      JmespathFreeRetryFeature.new
    when "test"
      JmespathFreeTestFeature.new
    when "timeout"
      JmespathFreeTimeoutFeature.new
    else
      JmespathFreeBaseFeature.new
    end
  end
end
